### PR TITLE
cpu/atmega*: increase default stack for "big" platforms

### LINKED ISSUE #9001: examples/gnrc_minimal: don't work anymore for atmega
#### Description

While doing some tests for the release, I noticed that the gnrc_minimal example
doesn't work anymore in AVR platforms. I tested for the 2017.10 release and
it works. I tried to bisect and fell into the netif2 commits, and git bisect cannot
help there anymore. It is broken since 2018.01.

#### Steps to reproduce the issue

Compile and test gnrc_minimal for any atmega platform. I'm using right now the
xbee and will test with other radios. The only I have currently is the mrf24j40.

#### Expected results

The example prints the address and you can ping it.

#### Actual results

The address is printed but you cannot ping it. I also put a `while (1) {}` which is
needed to avoid the node to kill all processes, as stated in #6526 .

#### Versions

Installed compiler toolchains 
-----------------------------
             native gcc: Apple LLVM version 8.0.0 (clang-800.0.42.1)
      arm-none-eabi-gcc: arm-none-eabi-gcc (GNU Tools for Arm Embedded Processors 7-2017-q4-major) 7.2.1 20170904 (release) [ARM/embedded-7-branch revision 255204]
                avr-gcc: avr-gcc (GCC) 7.3.0
       mips-mti-elf-gcc: missing
             msp430-gcc: msp430-gcc (GCC) 4.7.0 20120322 (mspgcc dev 20120911)
   riscv-none-embed-gcc: missing
                  clang: Apple LLVM version 8.0.0 (clang-800.0.42.1)

Installed compiler libs
-----------------------
   arm-none-eabi-newlib: "2.5.0"
    mips-mti-elf-newlib: missing
riscv-none-embed-newlib: missing
               avr-libc: "2.0.0" ("20150208")

Installed development tools
---------------------------
                  cmake: cmake version 3.11.0
               cppcheck: missing
                doxygen: missing
                 flake8: missing
                    git: git version 2.17.0
             coccinelle: missing

I have tagged it for the release just to not forget to put it as a known issue.

### PR DESCRIPTION
**Pull request by kYc0o**
### Contribution description
I had some trouble finding some networking "errors" when using atmega boards.
The issue was too small default stacks, which I consider can be increased for 
"big" atmega platforms.

### Issues/PRs references
Fixes #9001

### CODE REVIEW

File: cpu/atmega256rfr2/include/cpu_conf.h [hunk 1]
```diff
@@ -43,7 +43,7 @@ extern "C" {
 #define THREAD_STACKSIZE_DEFAULT   (512)
 #endif
 
-#define THREAD_STACKSIZE_IDLE      (129)
+#define THREAD_STACKSIZE_IDLE      (128)
```
Comment (bergzand) [inline cpu/atmega256rfr2/include/cpu_conf.h]: This is not an increase? Is this is a typo fix?

Comment (kYc0o) [reply cpu/atmega256rfr2/include/cpu_conf.h]: Oh, yes, should it go in it's own commit do you think?

Comment (bergzand) [reply cpu/atmega256rfr2/include/cpu_conf.h]: I don't mind, I was just curious

**Comment by kaspar030** [comment]
I'm not sure we're not running into trouble if we selectively increase the default stacksize on some of the avr8 boards. The results are not compile-checked.

I'd rather find a way of categorizing AVR boards and set the default stacksize based on that category.

**Comment by kYc0o** [comment]
Well, I just selected them based on their RAM. I don't know if there's an automatic way to know whenever an AVR cpu has more or less memory, it feels a bit arbitrary (besides the fact that the bigger the number model is, the bigger the memory is, but it's not 100% verified).

I doubled the size only for platforms with > 8K of RAM.

**Comment by kaspar030** [comment]
> Well, I just selected them based on their RAM.

Why don't you double them *all*? I think it is very important that the stacksize defaults are *per platform*, not per available RAM.

**Comment by kYc0o** [comment]
> Why don't you double them all?

Mostly, and I'd say only, because of the atmega328p, which has only 2K of RAM.

Currently every AVR platform defines its own stack. I don't mind to recap them into atmega_common, I'm just not sure about the smallest CPU. However, I don't know what's the current usage of that one if not only for educational purposes (it's also a challenging platform since we might want to run on it a lot of stuff knowing its tiny size).

**Comment by kaspar030** [comment]
> Mostly, and I'd say only, because of the atmega328p, which has only 2K of RAM.

Thing is, THREAD_STACKSIZE_DEFAULT is not for setting a sensible default stack size for a given CPU plus RAM size, but for setting the stacksize that makes *every module using the define* work properly. And as stack usage is *per architecture*, all AVR8 need to use the same default settings.

That way, applications might not link for atmega328p for lack of memory. That is much preferable to having them all compile and then fail at run-time.

Another story is that we might need more THREAD_STACKSIZE_* options (#1976), and possible also some way to have that information in the build system...
If you've identified that on AVR8, some modules need more stack than

**Comment by kYc0o** [comment]
Agreed, I'll unify them in atmega_common as other CPUs do.

**Comment by kaspar030** [comment]
That's gonna be a new PR then? If not, please reopen.

**Comment by gebart** [comment]
@kYc0o Another suggestion for the atmega is to move all of the cpu initialization to the common dir, all of the cpu.c files for the atmegas are functionally identical, only some editorial differences.

**Comment by kYc0o** [comment]
@gebart I'm exactly doing that, there are some things that need to be reused everywhere.

**Comment by ZetaR60** [comment]
@kYc0o Could this be reopened since the replacement (#9130) is blocked by the I2C work?

**Comment by kYc0o** [comment]
@ZetaR60 unfortunately I think #9130 doesn't solve the issue completely, since the goal is to unify the stack sizes for AVR8. Maybe the fix would be to assign other stack size to the threads that need it and thus solve #9001 .

**Pull request closed without merging**